validphys2/src/validphys/pineparser.py

```diff
@@ -300,11 +238,19 @@ def pineappl_reader(fkspec):
     sigma = pd.concat(partial_fktables, sort=True, copy=False).fillna(0.0)
 
     # Check whether this is a 1-point normalization fktable and, if that's the case, protect!
-    if fkspec.metadata.operation == "RATIO" and ndata == 1 and len(pines) == 1:
+    if fkspec.metadata.operation == "RATIO" and len(pines) == 1:
         # it _might_ be, check whether it is the divisor fktable
         divisor = fkspec.metadata.FK_tables[-1][0]
         name = fkspec.fkpath[0].name.replace(f".{EXT}", "")
-        protected = divisor == name
+
+        if np.allclose(sigma.loc[1:], 0.0):
+            # For old fktables, maybe the rest is filled with 0s
+            sigma = sigma.loc[0:0]
```

Comment:
Could you add a comment about this syntax (i.e. making sure sigma remains a dataframe)